### DISCUSSION TITLE
Android entry remove bottom line

### DISCUSSION THREAD

**Discussion by FM1973** [dotnet/maui#11247] [Q&A]
Hello.

.net 7 brought back the bottom line on android entries.
Our customer doesn´t like that.
Is there a way to remove the bottom line or at least set it´s color to transparent?

Thanks for your help!

**Comment by stevegdavis** [comment]
You could try handlers?, but this may mean that some of the functionality would not be available.

**Comment by hartez** [comment]
You can modify how the EntryHandler creates EditText controls - put something like this in your MauiProgram.cs:

```csharp
#if __ANDROID__
EntryHandler.PlatformViewFactory = (h) => {
	var editText = new AndroidX.AppCompat.Widget.AppCompatEditText(h.Context);
	editText.Background = null;
	editText.SetBackgroundColor(Android.Graphics.Color.Transparent);
	return editText;
};
#endif
```

**Comment by FM1973** [comment]
Thanks for your help.
I did the following in MauiProgram.cs

`private static void SetHandler()
    {

        Microsoft.Maui.Handlers.EntryHandler.Mapper.AppendToMapping("MyCustomization", (handler, view) =>
        {
            if (view is Entry)
            {

#if ANDROID
                        handler.PlatformView.Background = null;
#endif
            }
        });

    }`

This works. 
Thanks a lot!

**Comment by stevegdavis** [comment]
Net7 also now adds underline to a maui picker before it is clicked, any ideas? Thanks.
![Screenshot 2022-11-10 183705](https://user-images.githubusercontent.com/15140454/201178903-959f75a6-86bc-4ce1-a7f6-da23e6fa134d.png)

**Reply by hartez** [reply]
The picker control is built with an Entry (EditText) as the anchor, so you can apply a fix similar to the one above.

**Reply by stevegdavis** [reply]
@hartez Many thanks for the suggestion. I cannot see that the picker has an EditText anchored to it or am I missing something?

**Reply by hartez** [reply]
The MAUI Picker is rendered on Android as a MauiPicker, which is actually an AppCompatEditText: https://github.com/dotnet/maui/blob/0062faeba1dd391b1468798cd29299403b9b67f0/src/Core/src/Platform/Android/MauiPicker.cs

Because the MauiPicker is an AppCompatEditText, it gets the same underlining styling on Android as any other EditText control.

So you can modify that by doing something like this (not exact code, probably) in MauiProgram.cs:
```
#if __ANDROID__
PickerHandler.PlatformViewFactory = (h) => {
	var picker = new MauiPicker(h.Context);
	picker.Background = null;
	picker.SetBackgroundColor(Android.Graphics.Color.Transparent);
	return picker;
};
#endif
```

**Reply by stevegdavis** [reply]
Many thanks, works fine. I thought I had tried the MauiPicker, must have done something wrong as it didn't compile, oh well live and learn.
Thanks again.

**Reply by nnn149** [reply]
What if I only want to change a certain entry separately? Your method is globally effective. @hartez

**Reply by nnn149** [reply]
https://github.com/dotnet/maui/issues/7906#issuecomment-1577858789

**Reply by hartez** [reply]
> What if I only want to change a certain entry separately? Your method is globally effective. @hartez

Then you would need to subclass Entry and add a property to handle this. Or, alternatively, modify the EntryHandler mapping to check for a condition before applying the transparent background.

**Comment by KristyaIgor** [comment]
For me it will works with BackgroundTintList.

 Microsoft.Maui.Handlers.EditorHandler.Mapper.AppendToMapping("SetUpEntry", (handler, view) =>
            {
#if ANDROID
                handler.PlatformView.Background = null;
                handler.PlatformView.BackgroundTintList = ColorStateList.ValueOf(Android.Graphics.Color.Transparent);
#endif
            });
